src/services/fileStorage/utils/filePermissionHelper.js

```diff
@@ -7,60 +7,74 @@ const getFile = id => FileModel
 	.populate('owner')
 	.exec();
 
-const checkPermissions = (permission) => {
-	return async (user, file) => {
-		const fileObject = await getFile(file);
-		const {
-			permissions,
-			refOwnerModel,
-			owner: { _id: owner },
-		} = fileObject;
-
-		// return always true for owner of file
-		if (user.toString() === owner.toString()) {
-			return Promise.resolve(true);
-		}
+const checkMemberStatus = ({ file, user }) => {
+	const { owner: { userIds, teacherIds } } = file;
+	const finder = obj => user.equals(obj.userId || obj);
 
-		const isSubmission = await submissionModel.findOne({ fileIds: fileObject._id });
+	if (!userIds && !teacherIds) {
+		return false;
+	}
 
-		// or legacy course model
-		if (refOwnerModel === 'course' || isSubmission) {
-			const userObject = await userModel.findOne({ _id: user }).populate('roles').exec();
-			const isStudent = userObject.roles.find(role => role.name === 'student');
+	return userIds.find(finder) || teacherIds.find(finder);
+};
 
-			if (isStudent) {
-				const rolePermissions = permissions.find(
-					perm => perm.refId && perm.refId.toString() === isStudent._id.toString(),
-				);
-				return rolePermissions[permission] ? Promise.resolve(true) : Promise.reject();
-			}
-			return Promise.resolve(true);
-		}
+const checkPermissions = permission => async (user, file) => {
```

Comment:
I think the name `user` is a bit misguiding here, because it is just the user id